### PR TITLE
Fetch BSP environment for application runner

### PR DESCRIPTION
**Pull request by tpasternak**
Recently, we've added support for running Unit Tests with environment (classpath, java options, etc.) taken from BSP server. This feature currently supports ScalaTest runner and JUnit runner.

This PR add support for the Application Runner

Apart from that, commits cb83ef6 and 26225b1 contain some renaming and imporovements in error handling

### CODE REVIEW

File: bsp/resources/messages/ScalaBspBundle.properties [hunk 1]
```diff
@@ -74,13 +74,16 @@ bsp.test.no.class.matches.with.the.regex=No class matches with the regex
 bsp.task.fetching.sources=Fetching source list from BSP
 bsp.task.fetching.jvm.test.environment=Fetching JVM test environment from BSP
 bsp.task.error.could.not.extract.path=Could not extract project path from module {0}
+bsp.task.error.no.class.extractor="Could not find configuration extender for '{0}"
```
Comment (jastice) [inline bsp/resources/messages/ScalaBspBundle.properties]: quotes look wrong

File: bsp/resources/messages/ScalaBspBundle.properties [hunk 2]
```diff
@@ -74,13 +74,16 @@ bsp.test.no.class.matches.with.the.regex=No class matches with the regex
...
 bsp.task.error.could.not.fetch.sources=Could not fetch sources lists from BSP server. 'buildTarget/sources' request failed
 bsp.task.error.could.not.choose.any.target.id=Could not choose any target ID
 bsp.task.error.could.not.fetch.test.jvm.environment=Failed to fetch test JVM environment
 bsp.task.error=Error while fetching test environment from BSP: {0}
-bsp.task.error.test.env.not.supported=Build Server does not support 'buildTarget/jvmTestEnvironment' endpoint
+bsp.task.error.env.not.supported=Build Server does not support '${0}' endpoint
```
Comment (jastice) [inline bsp/resources/messages/ScalaBspBundle.properties]: no $ prefix in properties strings

File: bsp/resources/messages/ScalaBspBundle.properties (hunk 1 above)
Comment (tpasternak) [reply bsp/resources/messages/ScalaBspBundle.properties]: Thanks, I also noticed that the `{0}` tokens cannot be surrounded with `'`

File: bsp/resources/messages/ScalaBspBundle.properties (hunk 2 above)
Comment (tpasternak) [reply bsp/resources/messages/ScalaBspBundle.properties]: Done

**Review by jastice — APPROVED** [review]

**Comment by tpasternak** [comment]
Thanks for :) f40f2f2

**Review by jastice — APPROVED** [review]

File: bsp/resources/messages/ScalaBspBundle.properties [hunk 3]
```diff
@@ -10,10 +10,10 @@ bsp.widget.connections.on=on
 bsp.widget.connections.off=off
 
 ### org/jetbrains/bsp/data/dataObjects.scala
-bsp.metadata.error.project.info=BspMetadata.get: Could not get project info of project '{0}'
```
Comment (niktrop) [inline bsp/resources/messages/ScalaBspBundle.properties:13]: You may use double apostrophes to escape them.

**Comment by jastice** [comment]
rebased and merged in 9608c0a0243cec754e948e3bfa544cf4c6f67735

File: bsp/resources/messages/ScalaBspBundle.properties (hunk 3 above)
Comment (tpasternak) [reply bsp/resources/messages/ScalaBspBundle.properties:13]: Thanks! I'll do it later

**Pull request closed without merging**